### PR TITLE
Update Kotlin, Hilt, and all the dependencies. Fix upgrading issues

### PR DESCRIPTION
**Pull request by hrodrick**
#### ISSUE[#50]
* Update project dependencies

#### Description
* Updated Kotlin to 1.5.31
* Project dependencies updated to current versions. 
* Migrated Kotlin synthetics (`deprecated`) to android viewBinding
* Migrated deprecated & removed Hilt annotations to their equivalent in the new Hilt version. (`3.29.1`)
* Updated viewBinding and dataBinding configurations in app's build.gradle
* Replaced deprecated android.arch Room with AndroidX Room.
* Added "export=true" property on the manifest's launchable Activity because it is needed when targetting Android 12.
* Applied `kapt : correctErrorTypes` as it is required by new Hilt versions to correctly inject depedencies

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -4,35 +4,33 @@ apply plugin: 'kotlin-android'
 
 apply plugin: 'kotlin-kapt'
 
-apply plugin: 'kotlin-android-extensions'
-
```
Comment (amaury901130) [inline app/build.gradle:8]: why you remove this?

File: app/src/main/java/com/rootstrap/android/network/managers/ManagerModule.kt [hunk 2]
```diff
@@ -7,11 +7,11 @@ import com.rootstrap.android.network.managers.user.UserManagerImpl
 import dagger.Binds
 import dagger.Module
 import dagger.hilt.InstallIn
-import dagger.hilt.android.components.ApplicationComponent
+import dagger.hilt.components.SingletonComponent
 import javax.inject.Singleton
 
 @Module
-@InstallIn(ApplicationComponent::class)
+@InstallIn(SingletonComponent::class)
```
Comment (amaury901130) [inline app/src/main/java/com/rootstrap/android/network/managers/ManagerModule.kt:14]: could you please explain this change?

**Review by amaury901130 — APPROVED** [review]
Good, just a couple of questions.

File: app/build.gradle (hunk 1 above)
Comment (lmiotti) [reply app/build.gradle:8]: I think that `Kotlin extensions` are deprecated and replaced with `ViewBinding`. @hrodrick I'm right?

**Review by ximenaperez — APPROVED** [review]

File: app/src/main/java/com/rootstrap/android/network/providers/ServiceProviderModule.kt [hunk 3]
```diff
@@ -8,16 +8,16 @@ import com.squareup.moshi.Moshi
...
+import dagger.hilt.components.SingletonComponent
 import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
 import retrofit2.Retrofit
 import retrofit2.converter.moshi.MoshiConverterFactory
 import javax.inject.Singleton
 
 @Module
-@InstallIn(ApplicationComponent::class)
+@InstallIn(SingletonComponent::class)
```
Comment (lmiotti) [inline app/src/main/java/com/rootstrap/android/network/providers/ServiceProviderModule.kt:20]: Same as above

File: app/build.gradle (hunk 1 above)
Comment (marcossalto) [reply app/build.gradle:8]: Kotlin extension are deprecated. We should use view binding and @Parcelize. Good update!

File: .idea/gradle.xml [hunk 4]
```diff
@@ -15,7 +15,6 @@
           </set>
         </option>
         <option name="resolveModulePerSourceSet" value="false" />
-        <option name="useQualifiedModuleNames" value="true" />
       </GradleProjectSettings>
     </option>
   </component>
```
Comment (marcossalto) [inline .idea/gradle.xml:20]: I think this file should be ignore from git. Perhaps we need clear git cache to apply changes

**Review by marcossalto — APPROVED** [review]
Great work!

File: app/build.gradle (hunk 1 above)
Comment (hrodrick) [reply app/build.gradle:8]: Yes, Kotlin extensions are deprecated and kotlin synthetics is replaced by View & Data binding

File: app/src/main/java/com/rootstrap/android/network/managers/ManagerModule.kt (hunk 2 above)
Comment (hrodrick) [reply app/src/main/java/com/rootstrap/android/network/managers/ManagerModule.kt:14]: ApplicationComponent was renamed to SingletonComponent to allow the usage of Hilt in non-Android modules. You can check this release notes for further information : https://github.com/google/dagger/releases/tag/dagger-2.28.2

File: app/src/main/java/com/rootstrap/android/network/providers/ServiceProviderModule.kt (hunk 3 above)
Comment (hrodrick) [reply app/src/main/java/com/rootstrap/android/network/providers/ServiceProviderModule.kt:20]: ApplicationComponent was renamed to SingletonComponent to allow the usage of Hilt in non-Android modules. You can check this release notes for further information : https://github.com/google/dagger/releases/tag/dagger-2.28.2

File: .idea/gradle.xml (hunk 4 above)
Comment (hrodrick) [reply .idea/gradle.xml:20]: Yes you are right, I made the issue #52 to solve that problem.

**Pull request merged**